### PR TITLE
Make `decodeBitcoinAddress` throw on non-PKH addresses

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Refs: https://github.com/threshold-network/token-dashboard/issues/393

The `decodeBitcoinAddress` documentation says that this function decodes P2PKH and P2WPKH addresses into a public key hash. However, this requirement is not enforced by the function code. Here we add a validation that makes the function throw if the passed address is not PKH-based. That means the function will throw an exception if the passed address is, for example, P2SH or P2WSH.

### CODE REVIEW

File: typescript/src/bitcoin.ts [hunk 1]
```diff
@@ -389,13 +389,21 @@ export function encodeToBitcoinAddress(
...
  * @returns Public key hash decoded from the address. This will be an unprefixed
  *        hex string (without 0x prefix).
  */
 export function decodeBitcoinAddress(address: string): string {
   const addressObject = new bcoin.Address(address)
+
+  const isPKH =
+    addressObject.isPubkeyhash() || addressObject.isWitnessPubkeyhash()
+  if (!isPKH) {
+    throw new Error("address must be P2PKH or P2WPKH")
```
Comment (michalsmiarowski) [inline typescript/src/bitcoin.ts:404]: Non-blocking:

We usually start the error text with a capital letter in this project. 

Remember to also update the tests if you decide to fix that.

**Review by michalsmiarowski — APPROVED** [review]
LGTM 🔥 Left one non-blocking comments but overall looks good!

File: typescript/src/bitcoin.ts (hunk 1 above)
Comment (lukasz-zimnoch) [reply typescript/src/bitcoin.ts]: See https://github.com/keep-network/tbtc-v2/pull/510/commits/b446878f9dbe0a8bfaaf89b7d2c6508cb221d9f1

**Review by michalsmiarowski — APPROVED** [review]
LGTM 🔥 🚒

**Pull request merged**